avocado/core/runners/sysinfo.py

```diff
@@ -0,0 +1,311 @@
+import logging
+import multiprocessing
+import os
+import time
+import traceback
+from abc import ABC, abstractmethod
+
+from avocado.core import nrunner
+from avocado.core.runners.utils import messages
+from avocado.core.settings import settings
+from avocado.utils import astring, process, software_manager
+
+log = logging.getLogger("avocado.sysinfo")
+
+
+class _Collectible(ABC):
+
+    """
+    Abstract class for representing collectibles by sysinfo.
+    """
+
+    def __init__(self, logf):
+        self.logf = astring.string_to_safe_path(logf)
+
+    @abstractmethod
+    def run(self, queue, logdir):
+        pass
+
+
+class _Logfile(_Collectible):
+
+    """
+    Collectible system file.
+
+    :param path: Path to the log file.
+    """
+
+    def __init__(self, path):
+        super().__init__(os.path.basename(path))
+        self.path = path
+
+    def __repr__(self):
+        r = "sysinfo.Logfile(%r, %r)"
+        r %= (self.path, self.logf)
+        return r
+
+    def __eq__(self, other):
+        if isinstance(other, _Logfile):
+            return (self.path, self.logf) == (other.path, other.logf)
+        elif isinstance(other, _Collectible):
+            return False
+        return NotImplemented
+
+    def __ne__(self, other):
+        result = self.__eq__(other)
+        if result is NotImplemented:
+            return result
+        return not result
+
+    def __hash__(self):
+        return hash((self.path, self.logf))
+
+    def run(self, queue, logdir):
+        """
+        Send the content of log file to the runner messages queue.
+
+        :param queue: queue for the runner messages
+        :type queue: multiprocessing.SimpleQueue
+        :param logdir: Path to a log directory.
+        :type logdir: str
+        """
+        if os.path.exists(self.path):
+            try:
+                with open(self.path) as f:
+                    lines = f.readlines()
+                logf_path = os.path.join(logdir, self.logf)
+                for line in lines:
+                    queue.put(messages.FileMessage.get(line, logf_path))
+            except IOError:
+                log.debug("Not logging %s (lack of permissions)", self.path)
+        else:
+            log.debug("Not logging %s (file does not exist)", self.path)
+
+
+class _Command(_Collectible):
+
+    """
+    Collectible command.
+
+    :param cmd: String with the command.
+    :type cmd: str
+    """
+
+    def __init__(self, cmd):
+        super().__init__(cmd)
+        self.cmd = cmd
+
+    def __repr__(self):
+        return "sysinfo.Command(%r, %r)" % (self.cmd, self.logf)
+
+    def __eq__(self, other):
+        if isinstance(other, _Command):
+            return (self.cmd, self.logf) == (other.cmd, other.logf)
+        elif isinstance(other, _Collectible):
+            return False
+        return NotImplemented
```

Comment:
Based on  my previous comment, if this is part of the parent class, maybe we don't  need this  method.